zcash_client_sqlite/src/wallet.rs

```diff
@@ -3001,31 +2991,15 @@ pub(crate) fn store_decrypted_tx<P: consensus::Parameters>(
     // part) by this wallet.
     #[cfg(feature = "transparent-inputs")]
     if tx_has_wallet_outputs {
-        if let Some(b) = d_tx.tx().transparent_bundle() {
-            // queue the transparent inputs for enhancement
-            queue_tx_retrieval(
-                conn,
-                b.vin.iter().map(|txin| *txin.prevout.txid()),
-                Some(tx_ref),
-            )?;
-        }
+        queue_transparent_input_retrieval(conn, tx_ref, &d_tx)?
     }
 
     notify_tx_retrieved(conn, d_tx.tx().txid())?;
 
     // If the decrypted transaction is unmined and has no shielded components, add it to
     // the queue for status retrieval.
     #[cfg(feature = "transparent-inputs")]
-    {
-        let detectable_via_scanning = d_tx.tx().sapling_bundle().is_some();
-        #[cfg(feature = "orchard")]
-        let detectable_via_scanning =
-            detectable_via_scanning | d_tx.tx().orchard_bundle().is_some();
-
-        if d_tx.mined_height().is_none() && !detectable_via_scanning {
-            queue_tx_retrieval(conn, std::iter::once(d_tx.tx().txid()), None)?;
-        }
-    }
+    queue_unmined_tx_retrieval(conn, &d_tx)?;
```

Comment:
This is called whether or not the tx is mined. I suggest renaming it to something like `queue_tx_retrieval_if_needed`.